rust/rama/Cargo.toml

```diff
@@ -1,10 +1,11 @@
 [package]
 name = "server"
 version = "0.0.0"
-edition = "2021"
+edition = "2024"
```

Comment:
what does it change ?

I mean perhaps, we should replicate this for all rust

Reply:
https://github.com/the-benchmarker/web-frameworks/pull/8411#discussion_r2062562302

Reply:
Clear enough.

The it should not affect benchmark.

As I understand, it's more about features for `rama`. The should not this be driven be `rama` itself, not consumer app. I mean, when `rama` will use features from edition **2025**, will it be modified here ?

Reply:
Editions are only every 3 years, so that would be 2027.

It should not affect the benchmark no.

But because rama makes use of edition 2024 since a couple of versions ago, there is no way to update rama in a project as long as it's not also enabling edition 2024.

Editions are a mechanism in Rust to introduce breaking changes into the language without affecting existing libraries. E.g. there are still libraries out there on edition 2018 (the first edition). The key here is backwards compatible. Meaning that if a library is on edition 2024, you cannot use it as long as you are also not on edition 2024 (or later, which would be 2027 or 2030 or ...).

Read more: https://doc.rust-lang.org/edition-guide/editions/

Reply:
Ok, so it is a kind on minimal edition key.

Meaning that librairies in 2018 could be working in 2024, if no neww features of 2024 is used

Reply:
Yes and it is guaranteed those libraries on edition 2018 do not use features from edition 2021 / 2024 because they cannot do so, guaranteed by their defined edition. If you do not define edition it assumes edition 2018 (the original stable rust version basically).

However rama does already declare itself as edition 2024 since a couple of versions ago, so as a project (e.g. this benchmark rama binary project) you cannot use one of these latest rama versions if you do not also declare that you opt-in to edition 2024.